mesonbuild/interpreter.py

```diff
@@ -1214,6 +1214,60 @@ def get_cross_property_method(self, args, kwargs):
                 return args[1]
             raise InterpreterException('Unknown cross property: %s.' % propname)
 
+pch_kwargs = set(['c_pch', 'cpp_pch'])
+
+lang_arg_kwargs = set(['c_args',
+                       'cpp_args',
+                       'java_args',
+                       'rust_args',
+                       'objc_args',
+                       'objcpp_args',
+                       'fortran_args',
+                       'vala_args'])
+
+buildtarget_kwargs = set(['sources',
+                          'link_with',
+                          'link_whole',
+                          'link_args',
+                          'link_depends',
+                          'include_directories',
+                          'dependencies',
+                          'gui_app',
+                          'extra_files',
+                          'install',
+                          'install_rpath',
+                          'install_dir',
+                          'objects',
+                          'native',
+                          'name_suffix',
+                          'name_prefix',
+                          'build_by_default',
+                          'vs_module_defs',
+                          'pic',
+                          'override_options'])
+
+exe_kwargs = set()
+exe_kwargs.update(buildtarget_kwargs)
+exe_kwargs.update(lang_arg_kwargs)
+exe_kwargs.update(pch_kwargs)
+
+
+shlib_kwargs = set()
+shlib_kwargs.update(buildtarget_kwargs)
+shlib_kwargs.update(lang_arg_kwargs)
+shlib_kwargs.update(pch_kwargs)
+shlib_kwargs.update(['version', 'soversion'])
+shmod_kwargs = shlib_kwargs
+stlib_kwargs = shlib_kwargs
+
+permitted_kwargs = {'project': set(['version', 'meson_version', 'default_options', 'license', 'subproject_dir']),
+                    'executable': exe_kwargs,
+                    'shared_library': shlib_kwargs,
+                    'static_library': stlib_kwargs,
+                    'shared_module': shmod_kwargs,
+}
```

Comment:
__[flake8]__
* [E124] closing bracket does not match visual indentation

<!-- Comment by [SideCI](https://sideci.com) -->